Mathlib/Tactic/Linter/Header.lean

```diff
@@ -290,7 +299,15 @@ def duplicateImportsCheck (imports : Array Syntax)  : CommandElabM Unit := do
 @[inherit_doc Mathlib.Linter.linter.style.header]
 def headerLinter : Linter where run := withSetOptionIn fun stx ↦ do
   let mainModule ← getMainModule
-  unless Linter.getLinterValue linter.style.header (← getOptions) || (← isInMathlib mainModule) do
+  let inMathlib? := ← match ← InMathlibRef.get with
```

Comment:
Can you add a comment here, explaining that (a) the linter skips files not imported in `Mathlib.lean` to prevent warnings in "scratch files", and (b) this is an optimisation to avoid re-computing this information?